compiler/modules/front/codeGen.bal

```diff
@@ -1939,39 +1974,63 @@ final readonly & bir:UniformOrderType[] UNIFORM_ORDER_TYPES = [t:UT_BOOLEAN, t:U
 function operandOrderType(CodeGenContext cx, bir:Operand operand) returns bir:OrderType? {
     if operand is bir:Register {
         t:SemType operandTy = operand.semType;
-        if operandTy === t:NIL {
-            return ();
-        }
-        t:UniformTypeBitSet arrTy = t:LIST;
-        if t:isSubtypeSimple(operandTy, arrTy) {
-            t:UniformTypeBitSet? memberTy = t:simpleArrayMemberType(cx.mod.tc, operandTy);
-            if memberTy is t:UniformTypeBitSet {
-                bir:OrderType? ot = operandUniformOrderType(memberTy);
-                if ot is bir:UniformOrderType {
-                    return <bir:ArrayOrderType> [{opt:ot}];
-                }
-                if ot is bir:OptOrderType {
-                    return <bir:ArrayOrderType> [ot];
-                }
-            }
-            else {
-                panic err:impossible("Failed to get array member type");
-            }
-        }
-        else {
-            return operandUniformOrderType(operandTy);
-        }
+        return toOrderType(cx, operandTy);
     }
     else {
         var tc = t:constUniformTypeCode(operand);
-        if tc is bir:UniformOrderType {
+        if tc is t:UT_NIL {
+            return bir:nilOrderType;
+        }
+        else if tc is bir:UniformOrderType {
             return tc;
         }
     }
     return ();
 }
 
+function toOrderType(CodeGenContext cx, t:SemType operandTy) returns bir:OrderType? {
```

Comment:
operandOrderType goes from an Operand to an OrderType. This looks like it's going from a SemType to an OrderType, in which case a consistent name would be `semTypeOrderType`.